packages/api/src/external/commonwell-v2/document/cw-to-fhir.ts

```diff
@@ -117,3 +133,61 @@ function convertContextToFHIR(
     encounter: context.encounter ? [context.encounter] : undefined,
   }) as DocumentReferenceContext;
 }
+
+/**
+ * Type guard to check if an object is a valid FHIR Resource
+ */
+function isValidFHIRResource(obj: unknown): obj is Resource {
+  return (
+    typeof obj === "object" &&
+    obj !== null &&
+    "resourceType" in obj &&
+    typeof obj.resourceType === "string"
+  );
+}
+
+/**
+ * Safely converts a CommonWell contained resource to FHIR format.
+ * Validates the input before processing to prevent runtime errors.
+ *
+ * This only handles Organization resources for now, which is used on the Dashboard to specify
+ * the organization that created the document.
+ */
+function convertToFHIRResource(resource: unknown, patientId: string): Resource[] | undefined {
```

Comment:
How come a single resource gets transformed in many/array? It doesn't seem this is needed?